apache/superset
# feat(dashboard): Add thumbnails to dashboard edit draggable chart list

### SUMMARY

This PR adds thumbnails to the sidebar of draggable charts in the edit dashboard view and modifies the design of the cards with and without thumbnails enabled.  Thumbnails are shown when the `THUMBNAILS` feature flag is enabled.  If no thumbnail has been generated, the same placeholder that's used on the chart list card view is shown.  Other changes include:
- Layout updates for the text and the `Added` badge, when thumbnails are enabled and when they aren't.
- Text is truncated and shows a tooltip when it's too long to fit in the layout.
- The human-readable chart viz type is displayed instead of the machine-readable name.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


**Before:**

https://user-images.githubusercontent.com/13007381/178516532-8830f90c-24fd-441c-a864-47c74b618df8.mov

**After (thumbnails enabled):**

https://user-images.githubusercontent.com/13007381/178517080-023dd336-7c35-4784-9efb-28c822a1bf08.mov

**After (thumbnails disabled):**

https://user-images.githubusercontent.com/13007381/178517519-89811718-34e0-4a97-85f3-17f040885f25.mov

### TESTING INSTRUCTIONS


**Note:** Thumbnail generation is currently broken (see #20673).  In addition, there's a bug preventing the chart metadata from being downloaded (see #20684), so even if thumbnails are generated, any items in the draggable chart list with missing metadata won't show thumbnails either.

1. Run Superset with feature flag `THUMBNAILS=False` and ensure that the updated cards look and work as expected when added and not added to the dashboard. 
2. Run Superset with `THUMBNAILS=True` and ensure that thumbnails appear (or at least that their placeholders do) as expected and that the updated cards look and work as expected when added and not added to the dashboard.

### ADDITIONAL INFORMATION


- [x] Required feature flags: `THUMBNAILS`
